### PR TITLE
clean up

### PR DESCRIPTION
**Pull request by VictorVictini**

### CODE REVIEW

**Comment by VictorVictini** [comment]
might be a pattern here, maybe

**Comment by VictorVictini** [comment]
nothing should be broken, but double check just in case ig

**Review by mohfez — APPROVED** [review]
Tested the main functions, they all work good 😄 
Code also looks better 😀 

(lgtm 👍)

**Comment by mohfez** [comment]
maybe add a better readme before merging

File: api.go [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+		// the message was valid JSON
+		// unlock the mutex if the requested string has been returned by the server
+		if api.halted.on != "" && res["m"] == api.halted.on {
+			api.halted.mutex.Unlock()
+			api.halted.on = "" // resetting the mutex
+
+			// otherwise handle the response
+		} else {
+			api.handler(res, msg)
```
Comment (mohfez) [inline api.go:73]: Errors are not handled ? 😢

Comment (VictorVictini) [reply api.go:73]: damn I was rushing that one ig, I'll check it out in a bit

**Comment by VictorVictini** [comment]
readme added in main branch

**Pull request merged**